File: python-backend/backend/types/enums.py
"""Enumerations for the deep research agent v2."""

from enum import Enum


class ResearchStatus(str, Enum):
    PENDING = "pending"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"


class ConfidenceLevel(str, Enum):
    HIGH = "high"
    MEDIUM = "medium"
    LOW = "low"


class EventCategory(str, Enum):
    MODEL = "model"
    INFRA = "infra"
    MARKET = "market"
    REGULATION = "regulation"
    MOAT_ATTACK = "moat_attack"
    # Backward compat: old variants kept for deserialization of legacy reports
    PRODUCT_LAUNCH = "product_launch"
    FUNDING = "funding"
    PARTNERSHIP = "partnership"
    RESEARCH = "research"
    OPEN_SOURCE = "open_source"


class WhyIncludedTag(str, Enum):
    A = "A"
    B = "B"
    C = "C"
    D = "D"
    E = "E"
    F = "F"
    G = "G"

File: rust-deepagent-backend/src/types/enums.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum ResearchStatus {
    Pending,
    Running,
    Completed,
    Failed,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum ConfidenceLevel {
    High,
    Medium,
    Low,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum EventCategory {
    Model,
    Infra,
    Market,
    Regulation,
    MoatAttack,
    // Backward compat: old variants kept for deserialization of legacy reports
    #[serde(alias = "product_launch")]
    ProductLaunch,
    #[serde(alias = "funding")]
    Funding,
    #[serde(alias = "partnership")]
    Partnership,
    #[serde(alias = "research")]
    Research,
    #[serde(alias = "open_source")]
    OpenSource,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "UPPERCASE")]
pub enum WhyIncludedTag {
    #[serde(alias = "A")]
    A,
    #[serde(alias = "B")]
    B,
    #[serde(alias = "C")]
    C,
    #[serde(alias = "D")]
    D,
    #[serde(alias = "E")]
    E,
    #[serde(alias = "F")]
    F,
    #[serde(alias = "G")]
    G,
}
